rust/libnewsboat/src/configcontainer.rs

```diff
@@ -0,0 +1,1173 @@
+use std::collections::{HashMap, HashSet};
+use std::sync::{Arc, Mutex};
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum ConfigDataType {
+    Bool,
+    Int,
+    Enum,
+    Str,
+    Path,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum FeedSortMethod {
+    None,
+    FirstTag,
+    Title,
+    ArticleCount,
+    UnreadArticleCount,
+    LastUpdated,
+    LatestUnread,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum ArtSortMethod {
+    Title,
+    Flags,
+    Author,
+    Link,
+    Guid,
+    Date,
+    Random,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum SortDirection {
+    Asc,
+    Desc,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum ConfigHandlerStatus {
+    Success,
+    InvalidCommand,
+    TooFewParams,
+    TooManyParams,
+    InvalidParams(String),
+}
+
+#[derive(Clone, Debug)]
+pub struct ConfigData {
+    pub value: String,
+    pub default_value: String,
+    pub data_type: ConfigDataType,
+    pub enum_values: HashSet<String>,
+    pub multi_option: bool,
+}
+
+impl ConfigData {
+    pub fn new(val: &str, dtype: ConfigDataType) -> Self {
+        ConfigData {
+            value: val.to_string(),
+            default_value: val.to_string(),
+            data_type: dtype,
+            enum_values: HashSet::new(),
+            multi_option: false,
+        }
+    }
+
+    pub fn new_multi(val: &str, dtype: ConfigDataType) -> Self {
+        ConfigData {
+            value: val.to_string(),
+            default_value: val.to_string(),
+            data_type: dtype,
+            enum_values: HashSet::new(),
+            multi_option: true,
+        }
+    }
+
+    pub fn new_enum(val: &str, values: Vec<&str>) -> Self {
+        ConfigData {
+            value: val.to_string(),
+            default_value: val.to_string(),
+            data_type: ConfigDataType::Enum,
+            enum_values: values.iter().map(|s| s.to_string()).collect(),
+            multi_option: false,
+        }
+    }
+
+    pub fn set_value(&mut self, new_val: String) -> Result<(), String> {
+        match self.data_type {
+            ConfigDataType::Bool => {
+                if new_val == "true" || new_val == "yes" || new_val == "false" || new_val == "no" {
+                    self.value = new_val;
+                    Ok(())
+                } else {
+                    Err(format!("invalid boolean value: {new_val}"))
```

Comment:
In C++ code, this used to be:

```c++
			return nonstd::make_unexpected(
					strprintf::fmt(
						_("expected boolean value, found `%s' instead"),
						new_value));
```

The message changed here, and it isn't internationalized anymore.

Please see the output of `git grep gettext rust` to see how we handle i18n in Rust, and change the message back to what it was in C++ so gettext can find the existing translations.

(Same applies to messages further down below.)